src/Entities/ExerciseResult.cc

```diff
@@ -0,0 +1,46 @@
+#include "ExerciseResult.hh"
+
+#include <optional>
+
+namespace Entities
+{
+    Test::Test(const std::string &name, bool success)
+        : name_(name)
+    {
+        if (success)
+        {
+            errorMessage_ = std::nullopt;
+        }
+        else
+        {
+            errorMessage_ = "";
+        }
+    }
+
+    Test::Test(const std::string &name, const std::string &errorMessage)
+        : name_(name)
+    {
+        this->errorMessage_ = errorMessage;
+    }
+
+    bool Test::isSuccess() const
+    {
+        return errorMessage_ == std::nullopt;
```

Comment:
```suggestion
        return !errorMessage_;
```

Now that the string is an optional, I think this would work.

Reply:
It works, but I still prefer sth like !errorMessage.has_value() to make it more readable.